Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Middlewares/AmzSdkInvocationIdMiddleware.swift

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+import Smithy
+import SmithyHTTPAPI
+import struct Foundation.UUID
+
+private let AMZ_SDK_INVOCATION_ID_HEADER = "amz-sdk-invocation-id"
+
+/// Adds the amz-sdk-invocation-id header to requests.
+public struct AmzSdkInvocationIdMiddleware<InputType, OperationStackOutput> {
```

Comment:
is it weird that we're naming this middleware even though its acting as an interceptor? or is it actually a middleware we just don't need the middleware protocol anymore?